suse_migration_services/units/prepare.py

```diff
@@ -251,3 +262,29 @@ def get_root_disk_device(root_path):
                 block_type = block_record[1]
                 if block_type in considered_block_types:
                     return block_record[0]
+
+
+def get_regionsrv_client_file_location(root_path):
+    """
+    Find the correct path for the cloud-regionsrv-client cache files
+    as the location moves from /var/lib/cloudregister
+    (pre cloud-regionsrv-client-10.0.4) to /var/cache/cloudregister
+    (post cloud-regionsrv-client-10.0.4)
+    """
+    # cloud_register_path = os.sep.join(
+    #     [root_path, 'var', 'cache', 'cloudregister']
+    # )
+    for cloud_register_path in [
+        os.sep.join([root_path, 'var', 'cache', 'cloudregister']),
+        os.sep.join([root_path, 'var', 'lib', 'cloudregister'])
+    ]:
+        if os.path.isdir(cloud_register_path) and \
+                any('SMT' in x for x in os.listdir(cloud_register_path)):
+            return cloud_register_path
+        cloud_register_path = os.sep.join(
```

Comment:
The assignment is superfluous. If we do not find the cache files an exception is raised.

That said on a BYOS instance no such cache would exist and we would trigger the exception. Therefore we need to handle this in an intelligent way.

The problem is that we will have a chicken and egg problem the `registerutils.is_registered()` implementation needs the cache but we are trying to figure out whether the cache should exist and where it may be coming from. We can look at `/etc/hosts` to see if there is an entry for `.susecloud.net` if that exists it is fair to assume that the instance is connected to the update infrastructure and as such we need to look for the cache.

Reply:
@rjschwei  thanks I forgot to remove the comment and superfluous stuff after addressing @jesusbv feedback.  If I wrap this check around a check  for "/etc/hosts to see if there is an entry for .susecloud.net " is that sufficient?

Reply:
for example

```
    with open(hosts_setup, 'r', encoding="utf-8") as fp:
        if 'susecloud.net' in fp.read():
            try:
                cloud_register_metadata = \
                    get_regionsrv_client_file_location(root_path)
            except DistMigrationZypperMetaDataException as issue:
                log.error('Failed locating regionsrv-client cache files: {0}'.format(
                    issue)
                )
                raise
        else:
            cloud_register_metadata = os.sep.join(
                [root_path, 'var', 'lib', 'cloudregister']
            )
```

Reply:
@KeithMnemonic yes that's sufficient

Reply:
fixed